samples/CMakeLists.txt

```diff
@@ -95,83 +73,70 @@ if (CONFIG_SECURE_BOOT)
   endif()
 endif()
 
-# Special handling for the nRF53 when using RPMSG HCI.
-# Automatically include the multiprotocol_rpmsg as child image when both RPMSG HCI
-# and RPMSG 802.15.4 Serialization are enabled or the hci_rpmsg sample when
-# only RPMSG HCI is enabled. For any of these samples change the board to be the network core.
-if (CONFIG_BT_RPMSG_NRF53)
-  if (CONFIG_SOC_NRF5340_CPUAPP)
-
-    if (CONFIG_NRF_802154_SER_HOST)
-      set(NETCORE_IMAGE "multiprotocol_rpmsg")
-      set(NETCORE_IMAGE_PATH "${NRF_DIR}/samples/nrf5340/${NETCORE_IMAGE}")
-      message("Adding 'multiprotocol_rpmsg' firmware as child image since "
-      "CONFIG_BT_RPMSG_NRF53 and CONFIG_NRF_802154_SER_HOST are set to 'y'")
-    else()
-      set(NETCORE_IMAGE "hci_rpmsg")
-      set(NETCORE_IMAGE_PATH "${ZEPHYR_BASE}/samples/bluetooth/${NETCORE_IMAGE}")
-      message("Adding 'hci_rpmsg' firmware as child image since "
-      "CONFIG_BT_RPMSG_NRF53 is set to 'y'")
-    endif()
-
-    if (CONFIG_BT_LL_SOFTDEVICE_DEFAULT)
-      add_overlay_config(
-        ${NETCORE_IMAGE}
-        ${NRF_DIR}/subsys/bluetooth/controller/bt_ll_softdevice_hci_rpmsg.conf
-        )
-    endif()
+if (CONFIG_NCS_AUTO_INCLUDE_NET_CORE_CHILD_IMAGE
+    OR CONFIG_NCS_CHILD_IMAGE_SAMPLE_EMPTY_APP_CORE)
```

Comment:
I think you should keep the netcore and app core code paths completely separated.

That way it is much easier to read the code and know when code is working on app core specific images (such as the empty app), and netcore images such as the `RPMsg` samples.